From d744d6949646454a2de4c5269aee9023550666dd Mon Sep 17 00:00:00 2001
From: Antonios Motakis <a.motakis@virtualopensystems.com>
Date: Mon, 13 Oct 2014 14:06:16 +0100
Subject: [PATCH 108/132] iommu/arm-smmu: change IOMMU_EXEC to IOMMU_NOEXEC

commit a720b41c41f5a7e4c51558cf087882c57331581f upstream

Exposing the XN flag of the SMMU driver as IOMMU_NOEXEC instead of
IOMMU_EXEC makes it enforceable, since for IOMMUs that don't support
the XN flag pages will always be executable.

Signed-off-by: Antonios Motakis <a.motakis@virtualopensystems.com>
Acked-by: Joerg Roedel <jroedel@suse.de>
Signed-off-by: Will Deacon <will.deacon@arm.com>

[drivers/iommu/arm-smmu.c is not same as upstream, ignore its' changes.]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 include/linux/iommu.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index f5261fe..51ec18c 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -27,7 +27,7 @@
 #define IOMMU_READ	(1 << 0)
 #define IOMMU_WRITE	(1 << 1)
 #define IOMMU_CACHE	(1 << 2) /* DMA cache coherency */
-#define IOMMU_EXEC	(1 << 3)
+#define IOMMU_NOEXEC	(1 << 3)
 #define IOMMU_MMIO	(1 << 4) /* Device memory access */
 
 struct iommu_ops;
-- 
1.7.5.4

